samcli/commands/deploy/command.py

```diff
@@ -222,6 +234,14 @@ def do_cli(
         )
         guided_context.run()
 
+    if resolve_s3:
+        if s3_bucket or guided:
```

Comment:
I think if `resolve_s3` is true, and `--guided` is supplied, we don't need to error out and we can just continue. `--guided` implies this to be true.